overlord/snapstate/snapstate.go

```diff
@@ -64,7 +64,7 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		if !release.OnClassic {
 			return nil, fmt.Errorf("classic confinement is only supported on classic systems")
 		} else if !dirs.SupportsClassicConfinement() {
-			return nil, fmt.Errorf("classic confinement is not yet supported on your distribution")
+			return nil, fmt.Errorf(i18n.G("your distribution choose a directory layout incompatible with classic confinement. You can fix this yourself by setting a symlink from %s to /snap"), dirs.SnapMountDir)
```

Comment:
Perhaps a more neutral tone:

_"classic confinement requires snaps under /snap or symlink from /snap to %s"_

Note that the original order of directories seems reversed. The symlink has a "target", so _from_ the symlink _to_ the target seems more natural.